src/hooks/useAsset.tsx

```diff
@@ -0,0 +1,27 @@
+import { useRandomStorageProviderUrl } from '@/api/hooks'
+import { AssetAvailability, DataObject } from '@/api/queries'
+import { createStorageNodeUrl } from '@/utils/asset'
+
+export const useAsset = () => {
+  const randomStorageUrl = useRandomStorageProviderUrl()
+
+  const getAssetUrl = (availability?: AssetAvailability, assetUrls?: string[], dataObject?: DataObject | null) => {
+    if (availability !== AssetAvailability.Accepted) {
+      return
+    }
+    if (assetUrls?.length) {
+      return assetUrls[0]
+    }
+    if (!dataObject?.joystreamContentId) {
+      return
+    }
+    if (dataObject?.liaison?.isActive && dataObject?.liaison?.metadata) {
+      return createStorageNodeUrl(dataObject.joystreamContentId, dataObject?.liaison?.metadata)
+    }
+    if (!dataObject?.liaison?.isActive && randomStorageUrl) {
+      return createStorageNodeUrl(dataObject.joystreamContentId, randomStorageUrl)
+    }
+  }
+
+  return { getAssetUrl }
+}
```

Comment:
Not sure if it's enough for this task. It's kinda hard to test it because we have only one active worker with set metadata(URL).

Reply:
You can use the reckless deployment which has more providers for testing